Error when using with Webpack

I get this error with ember-auto-import@2.0.2 & Webpack@^5

```
Module not found: Error: Can't resolve 'fractal-page-object/dist/-private/types' in '/tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack'
resolve 'fractal-page-object/dist/-private/types' in '/tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack'
  Parsed request is a module
  No description file found in /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack or above
  resolve as module
    looking for modules in /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules
      existing directory /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object
        using description file: /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/package.json (relative path: .)
          using description file: /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/package.json (relative path: ./dist/-private/types)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/dist/-private/types doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/dist/-private/types.js doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/dist/-private/types.ts doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/dist/-private/types.json doesn't exist
            as directory
              /tmp/broccoli-443022G4IdTNSy0NUf/cache-199-webpack_bundler_ember_auto_import_webpack/node_modules/fractal-page-object/dist/-private/types doesn't exist
```

--- a/packages/ember-app/tests/integration/page-object-test.ts
+++ b/packages/ember-app/tests/integration/page-object-test.ts
@@ -3,6 +3,7 @@ import { setupRenderingTest } from 'ember-qunit';
 import { click, render } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 import { PageObject, selector } from 'fractal-page-object';
+import type { PageObjectConstructor } from 'fractal-page-object';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 
@@ -50,6 +51,35 @@ module('Integration | page object', function (hooks) {
 
     assert.equal(page2.element?.id, 'div1');
     assert.equal(page2[0].element?.id, 'div1');
+  });
+
+  test('can wrap selector()', async function (assert) {
+    const s = <T extends PageObject>(
+      name: string,
+      Cls?: PageObjectConstructor<T>
+    ): T => selector(`[data-name="${name}"]`, Cls);
+
+    await render(hbs`
+      <div data-name="one" class="div-one"/>
+      <div data-name="two" class="div-two">
+        <div data-name="three" class="div-three"/>
+      </div>
+    `);
+
+    const Page = class extends PageObject {
+      one = s('one');
+      two = s(
+        'two',
+        class extends PageObject {
+          three = s('three');
+        }
+      );
+    };
+    let page = new Page();
+
+    assert.dom(page.one.element).hasClass('div-one');
+    assert.dom(page.two.element).hasClass('div-two');
+    assert.dom(page.two.three.element).hasClass('div-three');
   });
 
   test('smoke test', async function (assert) {

--- a/packages/fractal-page-object/src/-private/factory.ts
+++ b/packages/fractal-page-object/src/-private/factory.ts
@@ -1,4 +1,4 @@
-import type { PageObjectClass } from './types';
+import type { PageObjectConstructor } from './types';
 import PageObject from '../page-object';
 
 /**
@@ -14,7 +14,10 @@ export default class PageObjectFactory<T extends PageObject> {
    * subclass of {@link PageObject}. If not supplied, {@link PageObject} will be
    * used.
    */
-  constructor(private selector: string, private Class?: PageObjectClass<T>) {}
+  constructor(
+    private selector: string,
+    private Class?: PageObjectConstructor<T>
+  ) {}
 
   /**
    * Create a {@link PageObject}
@@ -23,7 +26,7 @@ export default class PageObjectFactory<T extends PageObject> {
    * @returns the new page object
    */
   create(parent?: PageObject | Element): PageObject {
-    let Class = this.Class || (PageObject as PageObjectClass<PageObject>);
+    let Class = this.Class || (PageObject as PageObjectConstructor<PageObject>);
     return new Class(this.selector, parent);
   }
 }

--- a/packages/fractal-page-object/src/-private/global-factory.ts
+++ b/packages/fractal-page-object/src/-private/global-factory.ts
@@ -1,4 +1,4 @@
-import type { PageObjectClass } from './types';
+import type { PageObjectConstructor } from './types';
 import PageObject from '../page-object';
 import Factory from './factory';
 
@@ -22,7 +22,7 @@ export default class GlobalPageObjectFactory<
   constructor(
     selector: string,
     private rootElement?: Element,
-    Class?: PageObjectClass<T>
+    Class?: PageObjectConstructor<T>
   ) {
     super(selector, Class);
   }

--- a/packages/fractal-page-object/src/-private/helpers.ts
+++ b/packages/fractal-page-object/src/-private/helpers.ts
@@ -1,13 +1,15 @@
 import PageObject from '../page-object';
-import type { PageObjectClass } from './types';
+import type { PageObjectConstructor } from './types';
 
-function isPageObjectSubclass<T extends PageObject>(Class: PageObjectClass<T>) {
+function isPageObjectSubclass<T extends PageObject>(
+  Class: PageObjectConstructor<T>
+) {
   return Class === PageObject || Class.prototype instanceof PageObject;
 }
 
 export function validateSelectorArguments<T extends PageObject>(
   selector: string,
-  Class?: PageObjectClass<T>
+  Class?: PageObjectConstructor<T>
 ): void {
   if (!selector) {
     throw new Error('Cannot specify an empty selector');

--- a/packages/fractal-page-object/src/-private/types.ts
+++ b/packages/fractal-page-object/src/-private/types.ts
@@ -3,6 +3,9 @@ import type PageObject from '../page-object';
 export const DOM_QUERY = Symbol('DOM query');
 export const CLONE_WITH_INDEX = Symbol('withIndex');
 
-export type PageObjectClass<T extends PageObject> = new (
+/**
+ * A constructor for a {@link PageObject} or {@link PageObject} subclass
+ */
+export type PageObjectConstructor<T extends PageObject> = new (
   ...args: ConstructorParameters<typeof PageObject>
 ) => T | PageObject;

--- a/packages/fractal-page-object/src/global-selector.ts
+++ b/packages/fractal-page-object/src/global-selector.ts
@@ -1,13 +1,13 @@
 import GlobalPageObjectFactory from './-private/global-factory';
 import PageObject from './page-object';
-import type { PageObjectClass } from './-private/types';
+import type { PageObjectConstructor } from './-private/types';
 import { validateSelectorArguments } from './-private/helpers';
 
-type Arguments<T extends PageObject> = [string, PageObjectClass<T>?];
+type Arguments<T extends PageObject> = [string, PageObjectConstructor<T>?];
 type ArgumentsWithRoot<T extends PageObject> = [
   string,
   Element,
-  PageObjectClass<T>?
+  PageObjectConstructor<T>?
 ];
 
 /**
@@ -35,7 +35,7 @@ type ArgumentsWithRoot<T extends PageObject> = [
  */
 export default function globalSelector<T extends PageObject>(
   selector: string,
-  Class?: PageObjectClass<T>
+  Class?: PageObjectConstructor<T>
 ): T;
 
 /**
@@ -66,7 +66,7 @@ export default function globalSelector<T extends PageObject>(
 export default function globalSelector<T extends PageObject>(
   selector: string,
   rootElement: Element,
-  Class?: PageObjectClass<T>
+  Class?: PageObjectConstructor<T>
 ): T;
 
 /**

--- a/packages/fractal-page-object/src/index.ts
+++ b/packages/fractal-page-object/src/index.ts
@@ -1,4 +1,5 @@
 export { setRoot } from './-private/root';
+export type { PageObjectConstructor } from './-private/types';
 
 export { default as PageObject } from './page-object';
 

--- a/packages/fractal-page-object/src/page-object.ts
+++ b/packages/fractal-page-object/src/page-object.ts
@@ -3,7 +3,7 @@ import { getRoot } from './-private/root';
 import DOMQuery from './-private/dom-query';
 import createProxy from './-private/create-proxy';
 import { DOM_QUERY, CLONE_WITH_INDEX } from './-private/types';
-import type { PageObjectClass } from './-private/types';
+import type { PageObjectConstructor } from './-private/types';
 
 /**
  * This class implements all the basic page object functionality, and all page
@@ -141,7 +141,7 @@ export default class PageObject extends ArrayStub {
    * @private
    */
   [CLONE_WITH_INDEX](index: number): PageObject {
-    let Class = this.constructor as PageObjectClass<PageObject>;
+    let Class = this.constructor as PageObjectConstructor<PageObject>;
     return new Class('', this, index, this.rootElement);
   }
 

--- a/packages/fractal-page-object/src/selector.ts
+++ b/packages/fractal-page-object/src/selector.ts
@@ -1,6 +1,6 @@
 import PageObjectFactory from './-private/factory';
 import PageObject from './page-object';
-import type { PageObjectClass } from './-private/types';
+import type { PageObjectConstructor } from './-private/types';
 import { validateSelectorArguments } from './-private/helpers';
 
 /**
@@ -31,7 +31,7 @@ import { validateSelectorArguments } from './-private/helpers';
  */
 export default function selector<T extends PageObject>(
   selector: string,
-  Class?: PageObjectClass<T>
+  Class?: PageObjectConstructor<T>
 ): T {
   validateSelectorArguments(selector, Class);
 

